irslinger.h

```diff
@@ -52,6 +52,105 @@ static inline void gap(uint32_t outPin, double duration, gpioPulse_t *irSignal,
 	addPulse(0, 0, duration, irSignal, pulseCount);
 }
 
+// Transmit generated wave
+static inline int transmitWave(uint32_t outPin, gpioPulse_t *irSignal, unsigned int *pulseCount)
+{
+	// Init pigpio
+	if (gpioInitialise() < 0)
+	{
+		// Initialization failed
+		printf("GPIO Initialization failed\n");
+		return 1;
+	}
+
+	// Setup the GPIO pin as an output pin
+	gpioSetMode(outPin, PI_OUTPUT);
+
+	// Start a new wave
+	gpioWaveClear();
+
+	gpioWaveAddGeneric(*pulseCount, irSignal);
+	int waveID = gpioWaveCreate();
+
+	if (waveID >= 0)
+	{
+		int result = gpioWaveTxSend(waveID, PI_WAVE_MODE_ONE_SHOT);
+
+		printf("Result: %i\n", result);
+	}
+	else
+	{
+		printf("Wave creation failure!\n %i", waveID);
+	}
+
+	// Wait for the wave to finish transmitting
+	while (gpioWaveTxBusy())
+	{
+		time_sleep(0.1);
+	}
+
+	// Delete the wave if it exists
+	if (waveID >= 0)
+	{
+		gpioWaveDelete(waveID);
+	}
+
+	// Cleanup
+	gpioTerminate();
+	return 0;
+}
+
+static inline int irSlingRC5(uint32_t outPin,
+	int frequency,
+	double dutyCycle,
+	int pulseDuration,
+	const char *code)
+{
+	if (outPin > 31)
+	{
+		// Invalid pin number
+		return 1;
+	}
+
+	size_t codeLen = strlen(code);
+
+	printf("code size is %zu\n", codeLen);
+
+	if (codeLen > MAX_COMMAND_SIZE)
+	{
+		// Command is too big
+		return 1;
+	}
+
+	gpioPulse_t irSignal[MAX_PULSES];
+	int pulseCount = 0;
+
+	// Generate Code
+	int i;
+	for (i = 0; i < codeLen; i++)
+	{
+		if (code[i] == '0')
+		{
+			carrierFrequency(outPin, frequency, dutyCycle, pulseDuration, irSignal, &pulseCount);
+			gap(outPin, pulseDuration, irSignal, &pulseCount);
+		}
+		else if (code[i] == '1')
+		{
+			gap(outPin, pulseDuration, irSignal, &pulseCount);
+			carrierFrequency(outPin, frequency, dutyCycle, pulseDuration, irSignal, &pulseCount);
+		}
+		else
+		{
+			printf("Warning: Non-binary digit in command\n");
+		}
+	}
```

Comment:
I haven't looked super close, but can the RC5 protocol here be implemented in terms of the original `irSling` function? Something like setting `leadingPulseDuration = 0`, `leadingGapDuration = 0`, and `sendTrailingPulse  = false`?

Reply:
In the NEC protocol the difference between `0` and `1` is only the duration of the gap. RC-5 works in a completly other way: A logical `0` is a burst followed by a gap and a logical `1` is a gap followed by a burst. 
As far as I know, the code generator of `irSling` cannot create a logical `1` for RC-5 (gap followed by a burst) because in NEC all values are starting with a burst.

Reply:
Here you have a brief explanation of RC-5: https://www.sbprojects.net/knowledge/ir/rc5.php

Reply:
Ahhh I missed the fact the calls were swapped, I see it now. Thanks for the explanation!